```diff
diff --git a/sympy/stats/drv_types.py b/sympy/stats/drv_types.py
index d5a2c7b..1615a06 100644
--- a/sympy/stats/drv_types.py
+++ b/sympy/stats/drv_types.py
@@ -13,7 +13,7 @@
 
 from __future__ import print_function, division
 
-from sympy import (factorial, exp, S, sympify, And, I, zeta, polylog, log, beta, hyper, binomial,
+from sympy import (factorial, exp, S, sympify, I, zeta, polylog, log, beta, hyper, binomial,
                    Piecewise, floor)
 from sympy.stats import density
 from sympy.stats.drv import SingleDiscreteDistribution, SingleDiscretePSpace
@@ -43,7 +43,7 @@ class GeometricDistribution(SingleDiscreteDistribution):
 
     @staticmethod
     def check(p):
-        _value_check(And(0 < p, p <= 1), "p must be between 0 and 1")
+        _value_check((0 < p, p <= 1), "p must be between 0 and 1")
 
     def pdf(self, k):
         return (1 - self.p)**(k - 1) * self.p
@@ -115,7 +115,7 @@ class LogarithmicDistribution(SingleDiscreteDistribution):
 
     @staticmethod
     def check(p):
-        _value_check(And(p > 0, p < 1), "p should be between 0 and 1")
+        _value_check((p > 0, p < 1), "p should be between 0 and 1")
 
     def pdf(self, k):
         p = self.p
@@ -192,8 +192,8 @@ class NegativeBinomialDistribution(SingleDiscreteDistribution):
 
     @staticmethod
     def check(r, p):
-        _value_check((r > 0) != False, 'r should be positive')
-        _value_check(And(p > 0, p < 1), 'p should be between 0 and 1')
+        _value_check(r > 0, 'r should be positive')
+        _value_check((p > 0, p < 1), 'p should be between 0 and 1')
 
     def pdf(self, k):
         r = self.r
@@ -279,7 +279,7 @@ class PoissonDistribution(SingleDiscreteDistribution):
 
     @staticmethod
     def check(lamda):
-        _value_check((lamda > 0) != False, "Lambda must be positive")
+        _value_check(lamda > 0, "Lambda must be positive")
 
     def pdf(self, k):
         return self.lamda**k / factorial(k) * exp(-self.lamda)
@@ -369,7 +369,7 @@ class YuleSimonDistribution(SingleDiscreteDistribution):
 
     @staticmethod
     def check(rho):
-        _value_check((rho > 0) != False, 'rho should be positive')
+        _value_check(rho > 0, 'rho should be positive')
 
     def pdf(self, k):
         rho = self.rho
@@ -448,7 +448,7 @@ class ZetaDistribution(SingleDiscreteDistribution):
 
     @staticmethod
     def check(s):
-        _value_check((s > 1) != False, 's should be greater than 1')
+        _value_check(s > 1, 's should be greater than 1')
 
     def pdf(self, k):
         s = self.s
diff --git a/sympy/stats/frv_types.py b/sympy/stats/frv_types.py
index 75915ad..c6e3344 100644
--- a/sympy/stats/frv_types.py
+++ b/sympy/stats/frv_types.py
@@ -16,11 +16,10 @@
 from __future__ import print_function, division
 
 from sympy import (S, sympify, Rational, binomial, cacheit, Integer,
-        Dict, Basic, KroneckerDelta, Dummy, And, Eq)
+        Dict, Basic, KroneckerDelta, Dummy, Eq)
 from sympy.concrete.summations import Sum
 from sympy.core.compatibility import as_int, range
 from sympy.stats.rv import _value_check
-from sympy.core.logic import fuzzy_and, fuzzy_not
 from sympy.stats.frv import (SingleFinitePSpace, SingleFiniteDistribution)
 
 __all__ = ['FiniteRV', 'DiscreteUniform', 'Die', 'Bernoulli', 'Coin',
@@ -45,7 +44,7 @@ def dict(self):
     @staticmethod
     def check(density):
         for p in density.values():
-            _value_check(And(p >= 0, p <= 1),
+            _value_check((p >= 0, p <= 1),
                         "Probability at a point must be between 0 and 1.")
         _value_check(Eq(sum(density.values()), 1), "Total Probability must be 1.")
 
@@ -124,7 +123,7 @@ class DieDistribution(SingleFiniteDistribution):
 
     @staticmethod
     def check(sides):
-        _value_check(fuzzy_and([sides.is_positive, sides.is_integer]),
+        _value_check((sides.is_positive, sides.is_integer),
                     "number of sides must be a positive integer.")
 
     @property
@@ -178,7 +177,7 @@ class BernoulliDistribution(SingleFiniteDistribution):
 
     @staticmethod
     def check(p, succ, fail):
-        _value_check(And(p >= 0, p <= 1),
+        _value_check((p >= 0, p <= 1),
                     "p should be in range [0, 1].")
 
     @property
@@ -259,9 +258,9 @@ class BinomialDistribution(SingleFiniteDistribution):
 
     @staticmethod
     def check(n, p, succ, fail):
-        _value_check(fuzzy_and([n.is_integer, n.is_nonnegative]),
+        _value_check((n.is_integer, n.is_nonnegative),
                     "'n' must be nonnegative integer.")
-        _value_check((p <= 1) != False and (p >= 0) != False,
+        _value_check((p <= 1, p >= 0),
                     "p should be in range [0, 1].")
 
     @property
diff --git a/sympy/stats/joint_rv_types.py b/sympy/stats/joint_rv_types.py
index b973729..7b91f6e 100644
--- a/sympy/stats/joint_rv_types.py
+++ b/sympy/stats/joint_rv_types.py
@@ -1,4 +1,4 @@
-from sympy import (sympify, S, pi, sqrt, exp, Lambda, Indexed, Gt,
+from sympy import (sympify, S, pi, sqrt, exp, Lambda, Indexed,
     IndexedBase)
 from sympy.matrices import ImmutableMatrix
 from sympy.matrices.expressions.determinant import det
@@ -79,7 +79,7 @@ def check(self, mu, sigma):
         _value_check(len(mu) == len(sigma.col(0)),
             "Size of the mean vector and covariance matrix are incorrect.")
         #check if covariance matrix is positive definite or not.
-        _value_check(all([Gt(i, 0) != False for i in sigma.eigenvals().keys()]),
+        _value_check((i > 0 for i in sigma.eigenvals().keys()),
             "The covariance matrix must be positive definite. ")
 
     def pdf(self, *args):
@@ -120,7 +120,7 @@ def check(self, mu, sigma):
         _value_check(len(mu) == len(sigma.col(0)),
             "Size of the mean vector and covariance matrix are incorrect.")
         #check if covariance matrix is positive definite or not.
-        _value_check(all([Gt(i, 0) != False for i in sigma.eigenvals().keys()]),
+        _value_check((i > 0 for i in sigma.eigenvals().keys()),
             "The covariance matrix must be positive definite. ")
 
     def pdf(self, *args):
@@ -155,7 +155,7 @@ def check(self, mu, sigma, v):
         _value_check(len(mu) == len(sigma.col(0)),
             "Size of the location vector and shape matrix are incorrect.")
         #check if covariance matrix is positive definite or not.
-        _value_check(all([Gt(i, 0) != False for i in sigma.eigenvals().keys()]),
+        _value_check((i > 0 for i in sigma.eigenvals().keys()),
             "The shape matrix must be positive definite. ")
 
     def pdf(self, *args):
diff --git a/sympy/stats/rv.py b/sympy/stats/rv.py
index 413dbb0..756b6f2 100644
--- a/sympy/stats/rv.py
+++ b/sympy/stats/rv.py
@@ -1273,9 +1273,59 @@ def __getattr__(self, attr):
 
 def _value_check(condition, message):
     """
-    Check a condition on input value.
+    Raise a ValueError with message if condition is False, else
+    return True if all conditions were True, else False.
 
-    Raises ValueError with message if condition is not True
+    Examples
+    ========
+
+    >>> from sympy.stats.rv import _value_check
+    >>> from sympy.abc import a, b, c
+    >>> from sympy import And, Dummy
+
+    >>> _value_check(2 < 3, '')
+    True
+
+    Here, the condition is not False, but it doesn't evaluate to True
+    so False is returned (but no error is raised). So checking if the
+    return value is True or False will tell you if all conditions were
+    evaluated.
+
+    >>> _value_check(a < b, '')
+    False
+
+    In this case the condition is False so an error is raised:
+
+    >>> r = Dummy(real=True)
+    >>> _value_check(r < r - 1, 'condition is not true')
+    Traceback (most recent call last):
+    ...
+    ValueError: condition is not true
+
+    If no condition of many conditions must be False, they can be
+    checked by passing them as an iterable:
+
+    >>> _value_check((a < 0, b < 0, c < 0), '')
+    False
+
+    The iterable can be a generator, too:
+
+    >>> _value_check((i < 0 for i in (a, b, c)), '')
+    False
+
+    The following are equivalent to the above but do not pass
+    an iterable:
+
+    >>> all(_value_check(i < 0, '') for i in (a, b, c))
+    False
+    >>> _value_check(And(a < 0, b < 0, c < 0), '')
+    False
     """
-    if condition == False:
+    from sympy.core.compatibility import iterable
+    from sympy.core.logic import fuzzy_and
+    if not iterable(condition):
+        condition = [condition]
+    truth = fuzzy_and(condition)
+    if truth == False:
         raise ValueError(message)
+    return truth == True
```

This diff passes the stats tests.

--- a/sympy/core/tests/test_args.py
+++ b/sympy/core/tests/test_args.py
@@ -1246,7 +1246,8 @@ def test_sympy__stats__crv__ContinuousDistribution():
 
 def test_sympy__stats__frv_types__FiniteDistributionHandmade():
     from sympy.stats.frv_types import FiniteDistributionHandmade
-    assert _test_args(FiniteDistributionHandmade({1: 1}))
+    from sympy import Dict
+    assert _test_args(FiniteDistributionHandmade(Dict({1: 1})))
 
 
 def test_sympy__stats__crv__ContinuousDistributionHandmade():

--- a/sympy/stats/drv_types.py
+++ b/sympy/stats/drv_types.py
@@ -13,7 +13,7 @@ Zeta
 
 from __future__ import print_function, division
 
-from sympy import (factorial, exp, S, sympify, And, I, zeta, polylog, log, beta, hyper, binomial,
+from sympy import (factorial, exp, S, sympify, I, zeta, polylog, log, beta, hyper, binomial,
                    Piecewise, floor)
 from sympy.stats import density
 from sympy.stats.drv import SingleDiscreteDistribution, SingleDiscretePSpace
@@ -43,7 +43,7 @@ class GeometricDistribution(SingleDiscreteDistribution):
 
     @staticmethod
     def check(p):
-        _value_check(And(0 < p, p <= 1), "p must be between 0 and 1")
+        _value_check((0 < p, p <= 1), "p must be between 0 and 1")
 
     def pdf(self, k):
         return (1 - self.p)**(k - 1) * self.p
@@ -115,7 +115,7 @@ class LogarithmicDistribution(SingleDiscreteDistribution):
 
     @staticmethod
     def check(p):
-        _value_check(And(p > 0, p < 1), "p should be between 0 and 1")
+        _value_check((p > 0, p < 1), "p should be between 0 and 1")
 
     def pdf(self, k):
         p = self.p
@@ -193,7 +193,7 @@ class NegativeBinomialDistribution(SingleDiscreteDistribution):
     @staticmethod
     def check(r, p):
         _value_check(r > 0, 'r should be positive')
-        _value_check(And(p > 0, p < 1), 'p should be between 0 and 1')
+        _value_check((p > 0, p < 1), 'p should be between 0 and 1')
 
     def pdf(self, k):
         r = self.r

--- a/sympy/stats/frv.py
+++ b/sympy/stats/frv.py
@@ -191,6 +191,10 @@ class SingleFiniteDistribution(Basic, NamedArgsMixin):
         args = list(map(sympify, args))
         return Basic.__new__(cls, *args)
 
+    @staticmethod
+    def check(*args):
+        pass
+
     @property
     @cacheit
     def dict(self):

--- a/sympy/stats/frv_types.py
+++ b/sympy/stats/frv_types.py
@@ -16,34 +16,37 @@ Rademacher
 from __future__ import print_function, division
 
 from sympy import (S, sympify, Rational, binomial, cacheit, Integer,
-        Dict, Basic, KroneckerDelta, Dummy)
+        Dict, Basic, KroneckerDelta, Dummy, Eq)
 from sympy.concrete.summations import Sum
 from sympy.core.compatibility import as_int, range
-from sympy.core.logic import fuzzy_not, fuzzy_and
+from sympy.stats.rv import _value_check
 from sympy.stats.frv import (SingleFinitePSpace, SingleFiniteDistribution)
 
 __all__ = ['FiniteRV', 'DiscreteUniform', 'Die', 'Bernoulli', 'Coin',
         'Binomial', 'Hypergeometric']
 
 def rv(name, cls, *args):
-    density = cls(*args)
-    return SingleFinitePSpace(name, density).value
+    args = list(map(sympify, args))
+    i = 0
+    while i < len(args): # Converting to Dict since dict is not hashable
+        if isinstance(args[i], dict):
+            args[i] = Dict(args[i])
+        i += 1
+    dist = cls(*args)
+    dist.check(*args)
+    return SingleFinitePSpace(name, dist).value
 
 class FiniteDistributionHandmade(SingleFiniteDistribution):
     @property
     def dict(self):
         return self.args[0]
 
-    def __new__(cls, density):
-        density = Dict(density)
-        for k in density.values():
-            k_sym = sympify(k)
-            if fuzzy_not(fuzzy_and((k_sym.is_nonnegative, (k_sym - 1).is_nonpositive))):
-                raise ValueError("Probability at a point must be between 0 and 1.")
-        sum_sym = sum(density.values())
-        if sum_sym != 1:
-            raise ValueError("Total Probability must be equal to 1.")
-        return Basic.__new__(cls, density)
+    @staticmethod
+    def check(density):
+        for p in density.values():
+            _value_check((p >= 0, p <= 1),
+                        "Probability at a point must be between 0 and 1.")
+        _value_check(Eq(sum(density.values()), 1), "Total Probability must be 1.")
 
 def FiniteRV(name, density):
     """
@@ -118,12 +121,10 @@ def DiscreteUniform(name, items):
 class DieDistribution(SingleFiniteDistribution):
     _argnames = ('sides',)
 
-    def __new__(cls, sides):
-        sides_sym = sympify(sides)
-        if fuzzy_not(fuzzy_and((sides_sym.is_integer, sides_sym.is_positive))):
-            raise ValueError("'sides' must be a positive integer.")
-        else:
-            return super(DieDistribution, cls).__new__(cls, sides)
+    @staticmethod
+    def check(sides):
+        _value_check((sides.is_positive, sides.is_integer),
+                    "number of sides must be a positive integer.")
 
     @property
     @cacheit
@@ -174,14 +175,10 @@ def Die(name, sides=6):
 class BernoulliDistribution(SingleFiniteDistribution):
     _argnames = ('p', 'succ', 'fail')
 
-    def __new__(cls, *args):
-        p = args[BernoulliDistribution._argnames.index('p')]
-        p_sym = sympify(p)
-
-        if fuzzy_not(fuzzy_and((p_sym.is_nonnegative, (p_sym - 1).is_nonpositive))):
-            raise ValueError("p = %s is not in range [0, 1]." % str(p))
-        else:
-            return super(BernoulliDistribution, cls).__new__(cls, *args)
+    @staticmethod
+    def check(p, succ, fail):
+        _value_check((p >= 0, p <= 1),
+                    "p should be in range [0, 1].")
 
     @property
     @cacheit
@@ -259,18 +256,12 @@ def Coin(name, p=S.Half):
 class BinomialDistribution(SingleFiniteDistribution):
     _argnames = ('n', 'p', 'succ', 'fail')
 
-    def __new__(cls, *args):
-        n = args[BinomialDistribution._argnames.index('n')]
-        p = args[BinomialDistribution._argnames.index('p')]
-        n_sym = sympify(n)
-        p_sym = sympify(p)
-
-        if fuzzy_not(fuzzy_and((n_sym.is_integer, n_sym.is_nonnegative))):
-            raise ValueError("'n' must be positive integer. n = %s." % str(n))
-        elif fuzzy_not(fuzzy_and((p_sym.is_nonnegative, (p_sym - 1).is_nonpositive))):
-            raise ValueError("'p' must be: 0 <= p <= 1 . p = %s" % str(p))
-        else:
-            return super(BinomialDistribution, cls).__new__(cls, *args)
+    @staticmethod
+    def check(n, p, succ, fail):
+        _value_check((n.is_integer, n.is_nonnegative),
+                    "'n' must be nonnegative integer.")
+        _value_check((p <= 1, p >= 0),
+                    "p should be in range [0, 1].")
 
     @property
     @cacheit

--- a/sympy/stats/joint_rv_types.py
+++ b/sympy/stats/joint_rv_types.py
@@ -1,4 +1,4 @@
-from sympy import (sympify, S, pi, sqrt, exp, Lambda, Indexed, Gt,
+from sympy import (sympify, S, pi, sqrt, exp, Lambda, Indexed,
     IndexedBase)
 from sympy.matrices import ImmutableMatrix
 from sympy.matrices.expressions.determinant import det
@@ -79,7 +79,7 @@ class MultivariateNormalDistribution(JointDistribution):
         _value_check(len(mu) == len(sigma.col(0)),
             "Size of the mean vector and covariance matrix are incorrect.")
         #check if covariance matrix is positive definite or not.
-        _value_check(all([Gt(i, 0) != False for i in sigma.eigenvals().keys()]),
+        _value_check((i > 0 for i in sigma.eigenvals().keys()),
             "The covariance matrix must be positive definite. ")
 
     def pdf(self, *args):
@@ -120,7 +120,7 @@ class MultivariateLaplaceDistribution(JointDistribution):
         _value_check(len(mu) == len(sigma.col(0)),
             "Size of the mean vector and covariance matrix are incorrect.")
         #check if covariance matrix is positive definite or not.
-        _value_check(all([Gt(i, 0) != False for i in sigma.eigenvals().keys()]),
+        _value_check((i > 0 for i in sigma.eigenvals().keys()),
             "The covariance matrix must be positive definite. ")
 
     def pdf(self, *args):
@@ -155,7 +155,7 @@ class MultivariateTDistribution(JointDistribution):
         _value_check(len(mu) == len(sigma.col(0)),
             "Size of the location vector and shape matrix are incorrect.")
         #check if covariance matrix is positive definite or not.
-        _value_check(all([Gt(i, 0) != False for i in sigma.eigenvals().keys()]),
+        _value_check((i > 0 for i in sigma.eigenvals().keys()),
             "The shape matrix must be positive definite. ")
 
     def pdf(self, *args):

--- a/sympy/stats/rv.py
+++ b/sympy/stats/rv.py
@@ -1273,9 +1273,59 @@ class NamedArgsMixin(object):
 
 def _value_check(condition, message):
     """
-    Check a condition on input value.
+    Raise a ValueError with message if condition is False, else
+    return True if all conditions were True, else False.
 
-    Raises ValueError with message if condition is not True
+    Examples
+    ========
+
+    >>> from sympy.stats.rv import _value_check
+    >>> from sympy.abc import a, b, c
+    >>> from sympy import And, Dummy
+
+    >>> _value_check(2 < 3, '')
+    True
+
+    Here, the condition is not False, but it doesn't evaluate to True
+    so False is returned (but no error is raised). So checking if the
+    return value is True or False will tell you if all conditions were
+    evaluated.
+
+    >>> _value_check(a < b, '')
+    False
+
+    In this case the condition is False so an error is raised:
+
+    >>> r = Dummy(real=True)
+    >>> _value_check(r < r - 1, 'condition is not true')
+    Traceback (most recent call last):
+    ...
+    ValueError: condition is not true
+
+    If no condition of many conditions must be False, they can be
+    checked by passing them as an iterable:
+
+    >>> _value_check((a < 0, b < 0, c < 0), '')
+    False
+
+    The iterable can be a generator, too:
+
+    >>> _value_check((i < 0 for i in (a, b, c)), '')
+    False
+
+    The following are equivalent to the above but do not pass
+    an iterable:
+
+    >>> all(_value_check(i < 0, '') for i in (a, b, c))
+    False
+    >>> _value_check(And(a < 0, b < 0, c < 0), '')
+    False
     """
-    if condition == False:
+    from sympy.core.compatibility import iterable
+    from sympy.core.logic import fuzzy_and
+    if not iterable(condition):
+        condition = [condition]
+    truth = fuzzy_and(condition)
+    if truth == False:
         raise ValueError(message)
+    return truth == True

